bpfd/src/server/rpc.rs

```diff
@@ -171,23 +184,34 @@ impl Loader for BpfdLoader {
 
         // Await the response
         match resp_rx.await {
-            Ok(res) => {
-                let results = res.unwrap();
-                reply.xdp_mode = results.xdp_mode;
-                for r in results.programs {
-                    reply.results.push(ListResult {
-                        id: r.id,
-                        name: r.name,
-                        path: r.path,
-                        position: r.position as u32,
-                        priority: r.priority,
-                        proceed_on: r.proceed_on,
-                    })
+            Ok(res) => match res {
+                Ok(results) => {
+                    reply.xdp_mode = results.xdp_mode;
+                    for r in results.programs {
+                        reply.results.push(ListResult {
+                            id: r.id,
+                            name: r.name,
+                            path: r.path,
+                            position: r.position as u32,
+                            priority: r.priority,
+                            proceed_on: r.proceed_on,
+                        })
+                    }
+                    Ok(Response::new(reply))
                 }
-                Ok(Response::new(reply))
-            }
+                Err(e) => match e {
+                    BpfdError::NoProgramsLoaded => {
+                        warn!("{}", e);
```

Comment:
I don't think we need to `warn!` on this case as we're ignoring the error and returning a success response

Reply:
Good point, Just fixed this, thanks!